### ISSUE TITLE
PMP restarts when trying to use RRS on RasPi with MPEG4 video

### ISSUE THREAD

**Issue by OttoKerner** [plexinc/plex-media-player#7]
PMP-OE 1.0.0.4 
on RasPi 2

Refresh rate switching active, sync to video

tried to play a video with codec MPEG4/XVID and audio mp3 with constant frame rate @25 Hz

upon playback start, PMP switches screen to 1080p 50Hz and then restarts

```
<Media videoResolution="sd" id="711750" duration="2487080" bitrate="1500" width="720" height="400" aspectRatio="1.78" audioChannels="2" audioCodec="mp3" videoCodec="mpeg4" container="mkv" videoFrameRate="PAL" videoProfile="advanced simple" title="Original">
<Part accessible="1" exists="1" id="713058" key="/library/parts/713058/file.mkv" duration="2487080" file="redacted.mkv" size="466172004" container="mkv" indexes="sd" videoProfile="advanced simple">
<Stream id="910829" streamType="1" default="1" codec="mpeg4" index="0" bitrate="1214" bitDepth="8" bvop="1" chromaSubsampling="4:2:0" codecID="V_MS/VFW/FOURCC / XVID" duration="2487080" frameRate="25.000" frameRateMode="cfr" gmc="0" height="400" level="5" pixelFormat="yuv420p" profile="asp" qpel="0" scanType="progressive" width="720"/>
<Stream id="910830" streamType="2" default="1" codec="mp3" index="1" channels="2" bitrate="128" language="Deutsch" languageCode="ger" audioChannelLayout="stereo" bitrateMode="vbr" codecID="A_MPEG/L3" duration="2397649" samplingRate="48000"/>
<Stream id="910831" streamType="2" selected="1" codec="mp3" index="2" channels="2" bitrate="128" language="English" languageCode="eng" audioChannelLayout="stereo" bitrateMode="vbr" codecID="A_MPEG/L3" duration="2377656" samplingRate="48000"/>
</Part>
</Media>
```

last lines of the PMP log were
```
2015-10-20 20:37:20 [ INFO  ] DisplayManager.cpp @ 39 - "DisplayManager : Current Display Mode on Display #0 is  1920 x 1080 x 32bpp @50Hz" 
2015-10-20 20:37:20 [ INFO  ] DisplayComponent.cpp @ 78 - Trying to switch to best display mode. 
2015-10-20 20:37:20 [ INFO  ] DisplayComponent.cpp @ 157 - We think mode 7 is the best mode. 
2015-10-20 20:37:20 [ INFO  ] DisplayComponent.cpp @ 171 - Switching mode successful. 
2015-10-20 20:37:20 [ INFO  ] ComponentManager.cpp @ 39 - Component: display inited 
2015-10-20 20:37:20 [ INFO  ] ComponentManager.cpp @ 39 - Component: updater inited 
2015-10-20 20:37:20 [ INFO  ] ComponentManager.cpp @ 39 - Component: remote inited 
2015-10-20 20:37:20 [ DEBUG ] SettingsComponent.cpp @ 30 - Updated possible values for: "device" to (QVariant(QVariantMap, QMap(("title", QVariant(QString, "Autoselect device"))("value", QVariant(QString, "auto")))), QVariant(QVariantMap, QMap(("title", QVariant(QString, "bcm2835 ALSA, bcm2835 ALSA/Default Audio Device"))("value", QVariant(QString, "alsa/default:CARD=ALSA")))), QVariant(QVariantMap, QMap(("title", QVariant(QString, "bcm2835 ALSA, bcm2835 ALSA/Default Audio Device"))("value", QVariant(QString, "alsa/sysdefault:CARD=ALSA")))), QVariant(QVariantMap, QMap(("title", QVariant(QString, "bcm2835 ALSA, bcm2835 IEC958/HDMI/IEC958 (S/PDIF) Digital Audio Output"))("value", QVariant(QString, "alsa/iec958:CARD=ALSA,DEV=0")))), QVariant(QVariantMap, QMap(("title", QVariant(QString, "Default"))("value", QVariant(QString, "oss"))))) 
2015-10-20 20:37:20 [ DEBUG ] PlayerComponent.cpp @ 596 - Audio devices removed: QSet() 
2015-10-20 20:37:20 [ DEBUG ] PlayerComponent.cpp @ 597 - Audio devices added: QSet("alsa/sysdefault:CARD=ALSA", "alsa/default:CARD=ALSA", "alsa/iec958:CARD=ALSA,DEV=0", "auto", "oss") 
2015-10-20 20:37:20 [ DEBUG ] PlayerComponent.cpp @ 598 - Audio device selected: "auto" 
2015-10-20 20:37:20 [ INFO  ] PlayerComponent.cpp @ 725 - Audio Config - device: auto, channel layout: 2.0, passthrough codecs: ac3, ac3 transcoding: no 
2015-10-20 20:37:20 [ WARN  ] DisplayComponent.cpp @ 229 - Unable to locate current display. 
2015-10-20 20:37:20 [ INFO  ] ComponentManager.cpp @ 39 - Component: player inited 
2015-10-20 20:37:20 [ INFO  ] ComponentManager.cpp @ 39 - Component: power inited 
2015-10-20 20:37:20 [ INFO  ] ComponentManager.cpp @ 39 - Component: oesystem inited 
2015-10-20 20:37:20 [ DEBUG ] HelperLauncher.cpp @ 49 - Connecting to helper 
2015-10-20 20:37:20 [ DEBUG ] HelperLauncher.cpp @ 95 - Failed to connect to helper: "QLocalSocket::connectToServer: Connection refused" 
2015-10-20 20:37:20 [ DEBUG ] HelperLauncher.cpp @ 141 - Launching helper: "/usr/bin/pmphelper" 
2015-10-20 20:37:20 [ WARN  ] SettingsSection.cpp @ 69 - Looking for value: "fakefullscreen" in section: "main" but it can't be found 
2015-10-20 20:37:20 [ DEBUG ] KonvergoWindow.cpp @ 247 - FullScreen visbility set to  QWindow::Visibility(FullScreen) 
2015-10-20 20:37:20 [ DEBUG ] PowerComponent.cpp @ 76 - Disabling OS screensaver 
2015-10-20 20:37:20 [ FATAL ] main.cpp @ 62 - "EGL Error : Could not create the egl surface: error = 0x300b\n" 
```

The same video plays fine when RRS is off.

**Comment by wm4** [comment]
Has nothing to do with the codec. I'm pretty sure PMP just crashes sometimes after display modes are switched. (This is mostly a problem with Qt.)

The most interesting part is this log line:

>2015-10-20 20:37:20 [ FATAL ] main.cpp @ 62 - "EGL Error : Could not create the egl surface: error = 0x300b\n"

**Comment by tru** [comment]
This was fixed in 1.0.4
